api/scpca_portal/models/dataset.py

```diff
@@ -59,3 +61,69 @@ class Meta:
 
     def __str__(self):
         return f"Dataset {self.id}"
+
+    @staticmethod
+    def validate_data(json_obj: Dict) -> bool:
+        for project_id, config in json_obj.items():
+            data_element = DataElement(project_id, config)
+            if not data_element.validate():
+                return False
+
+        return True
+
+
+class DataElement:
+    def __init__(self, project_id: str, config: Dict) -> None:
+        self.project_id = project_id
+        self.config = config
+        self.merge_single_cell = self.config.get("merge_single_cell")
+        self.includes_bulk = self.config.get("includes_bulk")
+        self.SINGLE_CELL = self.config.get(Modalities.SINGLE_CELL)
+        self.SPATIAL = self.config.get(Modalities.SPATIAL)
+
+    def validate(self) -> bool:
+        return (
+            bool(self.project_id)
+            and self._validate_project_id()
+            and bool(self.config)
+            and len(self.config.keys()) == 4
+            and self._validate_merge_single_cell()
+            and self._validate_includes_bulk()
+            and self._validate_single_cell()
+            and self._validate_spatial()
+        )
```

Comment:
```suggestion
        @property
        def is_valid(self) -> bool:
            return all([self.validate_project(key for key in self.data.keys()])
            
        @property
        def valid_projects(): List[str]:
            return [key for key in self.data.keys() if self.validate_project(key) ]
            
        @property
        def invalid_projects(): List[str]
            pass #invert valid 
```

1. Let's change the `DataElement` to `DatasetData` and then create a property in dataset `dataset_data`.
2. Update validation to match above so that we have properties that can return actionable information and not just the boolean of is or isn't valid.
3. Let's internally throw errors inside of the individual checks but capture them inside of `validate_project`and for now we can always use the non-throwing interface.